# IS 362 - Week 5 Assignment

### Johnny Zgombic



### Step 1

The first step in our program is to import **pandas**

In [29]:
import pandas as pd

### Step 2

Set the default values for the maximum number of columns and rows returned.

In this case I chose to set the maximum columns and rows to 20. 

I chose 20 for the rows because this way if we sort, we will see the top and bottom 10 values.

In [30]:
pd.set_option('display.max_columns', 20)
pd.set_option('display.max_rows', 20)

### Step 3

Read the *airports.csv* file so that we can display the data using the **head()** function.

In [31]:
airports = pd.read_csv("airports.csv")
airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

### Step 4

Take the data we read and sort the values by **Latitude** in descending order.

In [32]:
airports.sort_values(by = "lat", ascending=False)

faa                                        name        lat         lon  \
417   EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   
982   NUI                             Nuiqsut Airport  70.210000 -151.005556   
1197  SCC                                   Deadhorse  70.194750 -148.465167   
232   BTI                          Barter Island Lrrs  70.133989 -143.581867   
1084  PIZ                              Point Lay Lrrs  69.732875 -163.005342   
...   ...                                         ...        ...         ...   
693   JHM                                     Kapalua  20.962936 -156.673031   
1006  OGG                                     Kahului  20.898650 -156.430458   
601   HNM                                        Hana  20.795636 -156.014439   
806   LNY                                       Lanai  20.785611 -156.951419   
1358  UPP                                       Upolu  20.265256 -155.859989   
930   MUE                               Waimea Kohala  20.001328 -155.668108   
1403  WKL                           Waikoloa Heliport  19.913600 -155.864000   
231   BSF                                Bradshaw Aaf  19.760056 -155.553717   
734   KOA                        Kona Intl At Keahole  19.738767 -156.045631   
679   ITO                                   Hilo Intl  19.721375 -155.048469   

       alt  tz dst              tzone  
417    149  -5   A                NaN  
230     44  -9   A  America/Anchorage  
110     41  -9   A  America/Anchorage  
708     35  -9   A  America/Anchorage  
152     96  -9   A  America/Anchorage  
1363    67  -9   A  America/Anchorage  
982     38  -9   A  America/Anchorage  
1197    64  -9   A  America/Anchorage  
232      2  -9   A  America/Anchorage  
1084    25  -9   A  America/Anchorage  
...    ...  ..  ..                ...  
693    256 -10   N   Pacific/Honolulu  
1006    54 -10   N   Pacific/Honolulu  
601     78 -10   A   Pacific/Honolulu  
806   1308 -10   N   Pacific/Honolulu  
1358    96 -10   A   Pacific/Honolulu  
930   2671 -10   A   Pacific/Honolulu  
1403   109 -10   N   Pacific/Honolulu  
231   6190 -10   A   Pacific/Honolulu  
734     47 -10   N   Pacific/Honolulu  
679     38 -10   N   Pacific/Honolulu  

[1458 rows x 8 columns]

#### QUESTION 1

###### What is the northernmost airport in the United States?

As far as **Latitude** is concerned, the higher the positive value, the further north we get. 

Looking at the data we compiled, the largest **Latitude** value is held by **Dillant Hopkins Airport**, however, when doing some research, the values for latitude and longitude for **Dillant Hopkins Airport** were reversed so that would make **Wiley Post-Will Rogers Memorial Airport** the northermost airport in the United States.

To verify this, I mapped the two airports and this confirmed my findings.

### Step 5

Take the data we read and sort the values by **Longitude** in descending order.

In [33]:
airports.sort_values(by = "lon", ascending=False)

faa                                 name        lat         lon   alt  \
1290  SYA                         Eareckson As  52.712275  174.113620    98   
942   MYF                     Montgomery Field  32.475900  117.759000    17   
396   DVT        Deer Valley Municipal Airport  33.411700  112.457000  1478   
417   EEN              Dillant Hopkins Airport  72.270833   42.898333   149   
444   EPM           Eastport Municipal Airport  44.910111  -67.012694    45   
624   HUL                         Houlton Intl  46.123083  -67.792056   489   
259   CAR                         Caribou Muni  46.871500  -68.017917   626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958  -68.044797   534   
1398  WFK  Northern Aroostook Regional Airport  47.285556  -68.312778   988   
192   BHB          Hancock County - Bar Harbor  44.449769  -68.361565    83   
...   ...                                  ...        ...         ...   ...   
1318  TNC                Tin City LRRS Airport  65.563056 -167.921667   269   
1389  WAA                        Wales Airport  65.622500 -168.095000    22   
652   IKO                 Nikolski Air Station  52.941667 -168.848889    77   
1055  PBV                            St George  56.578344 -169.661611   125   
1270  STG                   St. George Airport  56.577222 -169.663611   125   
1251  SNP                       St Paul Island  57.167333 -170.220444    63   
1281  SVA                     Savoonga Airport  63.686400 -170.493000    53   
522   GAM                      Gambell Airport  63.766800 -171.733000    27   
112   AKB                         Atka Airport  52.220278 -174.206389    56   
95    ADK                         Adak Airport  51.878000 -176.646000    18   

      tz dst              tzone  
1290  -9   A  America/Anchorage  
942    8   A     Asia/Chongqing  
396    8   A     Asia/Chongqing  
417   -5   A                NaN  
444   -5   A   America/New_York  
624   -5   A   America/New_York  
259   -5   A   America/New_York  
1101  -5   A   America/New_York  
1398  -5   A   America/New_York  
192   -5   A   America/New_York  
...   ..  ..                ...  
1318  -9   A  America/Anchorage  
1389  -9   A  America/Anchorage  
652   -9   A  America/Anchorage  
1055  -9   A  America/Anchorage  
1270  -9   A  America/Anchorage  
1251  -9   A  America/Anchorage  
1281  -9   A  America/Anchorage  
522   -9   A  America/Anchorage  
112   -9   A  America/Anchorage  
95    -9   A  America/Anchorage  

[1458 rows x 8 columns]

#### QUESTION 2

###### What is the easternmost airport in the United States?

As far as **Longitude** is concerned, the lower the positive value, the further east we get. 

Looking at the data we compiled, the smallest **Longitude** value is held by **Dillant Hopkins Airport**, however, as we saw in the last comparison, the values were switched.

Another issue that came into play is that when looking up **Eastport Municipal Airport**, the next lowest value, the value for **Longitude** should be positive, not negative. In either case, **Eastport Municipal Airport** is the easternmost airport in the United States.

### Step 6

Read the *weather.csv* file so that we can display the data using the **head()** function.

In [34]:
weather = pd.read_csv("weather.csv")
weather.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

### Step 7

Take the data we read and drill down into the values by so that we only display the **year** *2013*, the **month** *February* and the **day** *12*.

In [35]:
windiest_airport = weather[(weather['year'] == 2013) & (weather['month'] == 2) & (weather['day'] == 12)]
windiest_airport.sort_values('wind_speed', ascending = False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
9717     JFK  2013      2   12     8  42.08  24.08  48.56     280.0   
9716     JFK  2013      2   12     7  39.92  26.06  57.33     270.0   
18427    LGA  2013      2   12    12  44.06  23.00  43.02     270.0   
18416    LGA  2013      2   12     1  44.06  30.92  59.64     270.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
9732     JFK  2013      2   12    23  37.04  21.92  53.97     280.0   
9731     JFK  2013      2   12    22  37.94  21.92  52.09     290.0   
1007     EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1026     EWR  2013      2   12    20  37.94  24.08  57.04     280.0   
1028     EWR  2013      2   12    22  35.96  23.00  58.94     260.0   
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
9717     20.71404   27.61872     0.0    1011.6   10.0  2013-02-12T13:00:00Z  
9716     19.56326        NaN     0.0    1010.9   10.0  2013-02-12T12:00:00Z  
18427    19.56326   26.46794     0.0    1011.8   10.0  2013-02-12T17:00:00Z  
18416    19.56326   28.76950     0.0    1006.6   10.0  2013-02-12T06:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
9732      9.20624        NaN     0.0    1014.0   10.0  2013-02-13T04:00:00Z  
9731      9.20624        NaN     0.0    1013.4   10.0  2013-02-13T03:00:00Z  
1007      9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1026      9.20624        NaN     0.0    1013.9   10.0  2013-02-13T01:00:00Z  
1028      6.90468        NaN     0.0    1013.4   10.0  2013-02-13T03:00:00Z  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1027      5.75390        NaN     0.0    1013.0   10.0  2013-02-13T02:00:00Z  
18415     4.60312        NaN     0.0       NaN   10.0  2013-02-12T05:00:00Z  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
1029      4.60312        NaN     0.0    1013.7   10.0  2013-02-13T04:00:00Z  

[72 rows x 15 columns]

#### QUESTION 3

###### On February 12th, 2013, which New York area airport had the windiest weather?

As far as **Wind Speed** is concerned, the highest value wins. 

Looking at the data we compiled, the highest **Wind Speed** value is held by **EWR** at over 1,048.

Once again though, that value looks kind of unrealistic so I honestly would have to say that **LGA** was the windiest at just over 23, with a wind gust of just over 31.